src/ExpressionEvaluator/Core/Source/ResultProvider/Expansion/MemberExpansion.cs

```diff
@@ -437,24 +459,34 @@ internal StaticMembersExpansion(Type declaredType, Expansion members)
             MemberAndDeclarationInfo member,
             DkmClrValue memberValue,
             EvalResultDataItem parent,
+            DynamicFlagsMap dynamicFlagsMap,
             ExpansionFlags flags)
         {
-            var formatter = resultProvider.Formatter;
+            var declaredType = member.Type;
+            var declaredTypeInfo = dynamicFlagsMap.SubstituteDynamicFlags(member.OriginalDefinitionType, member.TypeInfo as bool[]);
```

Comment:
Would it make sense to do the substitution at the point the member is discovered? Would that mean we wouldn't need to pass around the dynamicFlagsMap, and potentially forget to substitute?